library/kotlin/src/io/envoyproxy/envoymobile/EnvoyStreamEmitter.kt

```diff
@@ -29,22 +29,14 @@ class EnvoyStreamEmitter(
     return this
   }
 
-  /**
-   * For ending an associated stream and sending trailers.
-   *
-   * @param trailers to send with ending a stream. If null, stream will be closed with an empty data frame.
-   */
-  override fun close(trailers: Map<String, List<String>>?) {
-    trailers?.let {
-      stream.sendTrailers(it)
-      return
-    }
-    stream.sendData(ByteBuffer.allocate(0), true)
+  override fun close(trailers: Map<String, List<String>>) {
+    stream.sendTrailers(trailers)
+  }
+
+  override fun close(byteBuffer: ByteBuffer) {
```

Comment:
What was the rationale for making this required (instead of having something like `fun close(byteBuffer: ByteBuffer = ByteBuffer.allocate(0)`

Reply:
It'd be ambiguous if we did that for trailers as well, right? Consumers calling `.close()` would no longer be able to differentiate between data and trailers

Reply:
Ah so the change here is to introduce the concept of closing with data versus closing with trailers

Reply:
Yes